timemachine/potentials/potential.py

```diff
@@ -48,6 +49,16 @@ def __call__(self, conf: Conf, box: Optional[Box]) -> float | Array:
     def to_gpu(self, precision: Precision) -> "BoundGpuImplWrapper":
         return self.potential.to_gpu(precision).bind(np.asarray(self.params))
 
+    def __post_init__(self):
+        # if isinstance(self.params, list) and len(self.params) == 0:
+        #     # untraced
+        #     self.params = np.array(self.params, dtype=np.float64)
+        # else:
+        #     print(self.params)
+        #     self.params = self.params.astype(np.float64)
+    
+        self.params = jnp.array(self.params, dtype=np.float64).reshape(self.potential.params_shape())
```

Comment:
@mcwitt I tried to post init all the params from bound potentials so that they're well-formed, well-shaped jax potentials, but this ended-up breaking all of the interpolation code

Reply:
eg: pytest -xsv tests/test_single_topology_combined.py::test_nonbonded_host_params_independent_of_lambda

results in:

```
tests/test_single_topology_combined.py:232:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_single_topology_combined.py:230: in get_nonbonded_host_params
    return st.combine_with_host(host_sys, lamb, num_water_atoms, st.ff, omm_topology).nonbonded_host.params
timemachine/fe/single_topology.py:1968: in combine_with_host
    guest_system = self.setup_intermediate_state(lamb=lamb)
timemachine/fe/single_topology.py:1722: in setup_intermediate_state
    proper = self.align_and_interpolate_propers(lamb)
timemachine/fe/single_topology.py:1659: in align_and_interpolate_propers
    return self._align_and_interpolate_bonded_term(
timemachine/fe/single_topology.py:1625: in _align_and_interpolate_bonded_term
    bonded_params.append(interpolate_fn(idxs, src_params, dst_params, lamb))
timemachine/fe/single_topology.py:1595: in _interpolate_torsion
    if self.all_idxs_belong_to_core(idxs):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <timemachine.fe.single_topology.SingleTopology object at 0x7fb02f8db460>, idxs = Traced<ShapedArray(float64[])>with<DynamicJaxprTrace(level=1/0)>

    def all_idxs_belong_to_core(self, idxs):
        core_atoms = self.get_core_atoms()
        print(core_atoms)
        print(idxs)
>       return all([x in core_atoms for x in idxs])
E       TypeError: iteration over a 0-d array
```